metanorm/normalizers/url.py

```diff
@@ -0,0 +1,201 @@
+                url_time, raw_attributes, ))
+            if not extracted_date:
+                return None
+            ########################################################################################
+            # further modification of "extracted time" based on other semantics of parts of the path
+            if raw_attributes['url'].startswith('ftp://ftp.remss.com'):
+                # 'd3d' cases are the average of three consequent days! so start day is yesterday!
+                # if condition is the search of 'd3d' in the filename
+                if 'd3d' in file_name:
+                    d = -1 if start else 1
+                    extracted_date = extracted_date + relativedelta(days=d)
+                # for weekly average ones in the "weeks" folder
+                # if condition is the search of "weeks" folder in the FTP path
+                elif "weeks" in url_path_and_file_name_splitted:
+                    d = -3 if start else 3
+                    extracted_date = extracted_date + relativedelta(days=d)
+                elif re.search(r"^f35_[0-9]{6}v[0-9]\.[0-9]\.gz$", file_name):
+                    # file is a month file.So,the end time must be the end of month.
+                    # python "strptime" always gives the first day. So the length of month in that
+                    # year is added.
+                    extracted_date = extracted_date if start else \
+                        extracted_date + self.length_of_month(extracted_date)
+            elif raw_attributes['url'].startswith('ftp://anon-ftp.ceda.ac.uk/neodc/esacci/sst/data/CDR_v2/Climatology/L4/v2.1'):
+                # the constant date is corrected based on the few letter at the beginning of file name
+                extracted_date += relativedelta(days=int(file_name[1:4]) - 1)
+            elif raw_attributes['url'].startswith('ftp://ftp.gportal.jaxa.jp/standard/GCOM-W/GCOM-W.AMSR2'):
+                if file_name_splitted[1].endswith('00'):  # it is a month file
+                    extracted_date = extracted_date if start else \
+                        extracted_date + self.length_of_month(extracted_date)
+            return extracted_date
+
+    def get_provider(self, raw_attributes):
+        """ returns the suitable provider based on the filename """
+        founded_values = self.find_matching_value(self.urls_provider, raw_attributes, )
+        if founded_values:
+            return pti.get_gcmd_provider(founded_values)
+
+    @staticmethod
+    def create_parameter_list(parameters):
+        """ Convert list with standard names into list with Pythesing dicts """
+        if parameters:
+            return [pti.get_cf_standard_name(cf_parameter) for cf_parameter in parameters]
+
+    def get_dataset_parameters(self, raw_attributes):
+        """ return list with different parameter(s) from cf_standard_name """
+        return \
+            self.create_parameter_list(self.find_matching_value(
```

Comment:
Why do you add a backslash here? It saves you only 3 characters. But `return  self.create_parameter_list(self.find_matching_value(` is still well below the limit!

```suggestion
        return self.create_parameter_list(self.find_matching_value(
```

Reply:
it is a combination of my correction and autopep8. I have correct it. Done in the next commit.